### DISCUSSION TITLE
Field Update By Changes of Another Field Defined Automatically

### DISCUSSION THREAD

**Discussion by ismaeltj9** [filamentphp/filament#12598] [Help] [answered]
### Package

Form builder

### Package Version

v3.2.70

### How can we help you?

This is how AfterStateUpdated() works, but what if I need to update by changes of another field that was automatically update by $set()? for examples I have price, quantity and total price in a Repeater, and a total of all itens?, how to calculate this? Thanks any help.
```
Field updates
You may use the afterStateUpdated() method to customize what happens after a field is updated by the user. Only changes from the user on the frontend will trigger this function, not manual changes to the state from $set() or another PHP function.
```

**Comment by leandrocfe** [comment, marked as answer]
```php
Forms\Components\Group::make([
    Forms\Components\Repeater::make('products')
        ->columns(3)
        ->schema([
            Forms\Components\TextInput::make('price')
                ->live(onBlur: true)
                ->required(),
            Forms\Components\TextInput::make('quantity')
                ->live(onBlur: true)
                ->required(),
            Forms\Components\Placeholder::make('total')
                ->content(fn(Forms\Get $get) => $get('price') * $get('quantity')),
        ]),
    Forms\Components\Placeholder::make('total price')
        ->content(function (Forms\Get $get) {
            $map = Arr::map($get('products'), function ($item) {
                return $item['price'] * $item['quantity'];
            });
            return array_sum($map);
        }),
])->columns(2),
```

**Reply by ismaeltj9** [reply]
Hello @leandrocfe , work very well, thank you again for another help!

**Reply by w2ca** [reply]
@leandrocfe tem como fazer isso porem com um TextInput?

**Reply by leandrocfe** [reply]
fazer oq?

**Reply by w2ca** [reply]
@leandrocfe  o mesmo que você fez com o Placeholder, porem com o TextInput.

**Reply by leandrocfe** [reply]
Como é TextInput, talvez tenha que usar Set $set

**Reply by ismaeltj9** [reply]
> @leandrocfe tem como fazer isso porem com um TextInput?

se for TextInput você pode usar a função hidden somente para executar alguma função, o hidden da mesma forma que o contente do placeholder é executado automaticamente quando alguma coisa é alterada, veja no exemplo que eu retorno false para não ter hidden mas eu executo um set dentro dele.

Forms\Components\TextInput::make('valor_conta')
    ->hidden(function (Get $get, Set $set) {
        $set('valor_conta', (float)$get('percentual') * (float)$get('../../valor_compra') / 100);
        return false;
    })
